File: php/src/Application/UseCase/CreateGreetingUseCase.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\UseCase;

use Alxarafe\App\Domain\Model\Greeting;
use Alxarafe\App\Domain\Model\GreetingRepositoryInterface;

final readonly class CreateGreetingUseCase
{
    public function __construct(
        private GreetingRepositoryInterface $repository,
    ) {
    }

    public function execute(string $name): Greeting
    {
        $id = bin2hex(random_bytes(16));
        $message = sprintf('Hello, %s!', $name);
        $createdAt = new \DateTimeImmutable();

        $greeting = new Greeting($id, $message, $createdAt);

        $this->repository->save($greeting);

        return $greeting;
    }
}

File: java/src/main/java/com/alxarafe/app/application/usecase/CreateGreetingUseCase.java
package com.alxarafe.app.application.usecase;

import com.alxarafe.app.domain.model.Greeting;
import com.alxarafe.app.domain.model.GreetingRepository;

import java.time.Instant;
import java.util.UUID;

public final class CreateGreetingUseCase {
    private final GreetingRepository repository;

    public CreateGreetingUseCase(GreetingRepository repository) {
        this.repository = repository;
    }

    public Greeting execute(String name) {
        String id = UUID.randomUUID().toString().replace("-", "");
        String message = "Hello, " + name + "!";
        Instant createdAt = Instant.now();

        Greeting greeting = new Greeting(id, message, createdAt);
        repository.save(greeting);

        return greeting;
    }
}
